catalog/controller/payment/razorpay.php

```diff
@@ -105,6 +105,24 @@ public function index(): string {
                 $data['is_recurring'] = "false";
                 // Orders API with payment autocapture
                 $order_data = $this->get_order_creation_data($this->session->data['order_id']);
+
+                // validate order amount
+                $this->load->model('checkout/cart');
+                $totals = [];
+                $taxes = $this->cart->getTaxes();
+                $total = 0;
+                ($this->model_checkout_cart->getTotals)($totals, $taxes, $total);
+                $cartTotal = $total * 100;
+
+                if ($order_data["amount"] != $cartTotal and

```

Comment:
@abdulwahidsharief , why this check ?

Reply:
@ramth05 we are not receiving updated order amount because of order_id is set in session so we are unsetting order_id if cart amount and order amount mismatches